assets/profiles/dependent_fields.js

```diff
@@ -1,12 +1,23 @@
 import $ from 'jquery';
 
 export default () => {
-  $('#field-of-study-application-documentation').hide();
+  let master_doc = $('#field-of-study-application-master-documentation');
+  let social_doc = $('#field-of-study-application-documentation');
+  social_doc.hide();
+  master_doc.hide();
+
   $('#id_field_of_study').on('change', () => {
-    if ($('#id_field_of_study').val() === '40') {
-      $('#field-of-study-application-documentation').show();
-    } else {
-      $('#field-of-study-application-documentation').hide();
+      let field_of_study = parseInt($('#id_field_of_study').val());
+
+    if (field_of_study >= 10 && field_of_study <= 30){
+      social_doc.hide();
+      master_doc.show();
+    } else if (field_of_study === 40){
+      social_doc.show();
+      master_doc.hide();
+    } else{
+      social_doc.hide();
+      master_doc.hide();
```

Comment:
Personal preference probably, but I'd rather hide it by default and show it conditionally than hide and or show it conditionally.

```javascript
social_doc.hide()
master_doc.hide()
if master study
    master_doc.show()
else if social study
    social_doc.show()
```

And maybe even hide the HTML element by default (using CSS) instead of doing two DOM operations which hide or show the element (jQ might be efficient enough to only be done in one operation, but I'm not sure).

```javascript
if master study
    master_doc.show()
else if social study
    social_doc.show()
```

Both ways are cleaner code nonetheless.